src/unity/lib/annotation/utils.cpp

```diff
@@ -1,10 +1,91 @@
 #include "utils.hpp"
+#include <logger/assertions.hpp>
+#include <toolkits/image_deep_feature_extractor/image_deep_feature_extractor_toolkit.hpp>
 
-bool turi::annotate::isInteger(std::string s) {
-	if(s.empty() || ((!isdigit(s[0])) && (s[0] != '-') && (s[0] != '+'))) return false;
+namespace turi {
+namespace annotate {
 
-	char * p;
-	strtol(s.c_str(), &p, 10);
+bool isInteger(std::string s) {
+  if (s.empty() || ((!isdigit(s[0])) && (s[0] != '-') && (s[0] != '+')))
+    return false;
 
-	return (*p == 0);
-}
\ No newline at end of file
+  char *p;
+  strtol(s.c_str(), &p, 10);
+
+  return (*p == 0);
```

Comment:
This code right here is wrong (see the definition of strtol).  It should be `return (*p == s.data() + s.size())`.

Reply:
Okay, to file an issue on this and ignore it here, but it should get fixed.

Reply:
I’m having a difficult time understanding the reasoning, here’s my understanding:

As it parses the string `strtol` will stop at the first character that’s not considered an integer. The  `p` provided then is populated with the pointer to the first non-integer character, hence in the case if a `NULL` terminated string is 0. if that character isn't 0 or `NULL` than we can safely assume that the string isn't a parsable integer.

-- Likewise--

`(s.data() + s.size())` - is the pointer to the last character of the string. so rather than comparing the dereferenced pointer `*p` we can compare it to the pointer itself `p == (s.data() + s.size())` and we get the same result.

One is a pointer level comparison, the other is a content level comparison.

What am I missing here?

Reply:
Ah, you're right.  Yes.